spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/LoadBalancerAutoConfiguration.java

```diff
@@ -79,26 +75,33 @@ public SmartInitializingSingleton loadBalancedRestTemplateInitializerDeprecated(
 		});
 	}
 
-	@Bean
-	public SmartInitializingSingleton loadBalancedRestClientBuilderInitializer(
-			ObjectProvider<List<RestClientBuilderCustomizer>> restClientBuilderCustomizers) {
-		return () -> restClientBuilderCustomizers.ifAvailable(customizers -> {
-			for (RestClient.Builder restClientBuilder : restClientBuilders) {
-				for (RestClientBuilderCustomizer customizer : customizers) {
-					customizer.customize(restClientBuilder);
-				}
-
-			}
-		});
-	}
-
 	@Bean
 	@ConditionalOnMissingBean
 	public LoadBalancerRequestFactory loadBalancerRequestFactory(LoadBalancerClient loadBalancerClient) {
 		return new LoadBalancerRequestFactory(loadBalancerClient, transformers);
 	}
 
-	@Configuration(proxyBeanMethods = false)
+	@AutoConfiguration
+	static class DeferringLoadBalancerInterceptorConfig {
```

Comment:
I assume these inner classes get autoconfigured because the outerclass is specified in spring.factories?